### PR TITLE
[charlierose] Add new extractor (closes #10133)

### LINKED ISSUE #10133: Add download video website: https://charlierose.com/videos/27996
Add download video website: https://charlierose.com/videos/27996

There's nothing download video tool can download video from https://charlierose.com/videos/27996, 
and many users need it . Pls add it .

Thanks a lot.

### PR DESCRIPTION
**Pull request by TRox1972**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This adds an extractor for charlierose.com

### CODE REVIEW

**Comment by yan12125** [comment]
Use ```self._parse_html5_media_entries``` instead.

**Comment by TRox1972** [comment]
@yan12125 does this look good?

**Comment by yan12125** [comment]
Well, since commit 520251c093f5e0fe6af5e57203a0452aef0682ac M3U8 manifests in HTML5 <video> tags are supported. Could you rebase this PR to the current git master?

**Comment by TRox1972** [comment]
rebased and updated

File: youtube_dl/extractor/charlierose.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    }
+
+    _PLAYER_BASE = 'https://charlierose.com/video/player/%s'
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(self._PLAYER_BASE % video_id, video_id)
+
+        title = remove_end(self._og_search_title(webpage, fatal=True), ' - Charlie Rose')
```
Comment (yan12125) [inline youtube_dl/extractor/charlierose.py]: ```_og_search_title``` has already defaulted to ```fatal=True```

Comment (TRox1972) [inline youtube_dl/extractor/charlierose.py]: fixed

**Comment by yan12125** [comment]
Thanks!

**Pull request merged**